src/renderer/components/deposit/withdraw/Withdraw.tsx

```diff
@@ -1,284 +1,339 @@
+    if (RD.isInitial(withdrawRD)) return <></>
+
+    // Get timer value
+    const timerValue = FP.pipe(
+      withdrawRD,
       RD.fold(
-        () => loading,
-        () => loading,
-        (error) => (
-          <>
-            {intl.formatMessage({ id: 'common.error' })} {error?.message ?? ''}
-          </>
+        () => 0,
+        FP.flow(
+          O.map(({ loaded }) => loaded),
+          O.getOrElse(() => 0)
         ),
-        ({ thorMemo, thorOut, assetOut }) => {
-          const formattedThorMemoFee = formatFee({
-            amount: thorMemo,
-            asset: AssetRuneNative
-          })
-          const formattedAssetOutFee = formatFee({
-            amount: calculateAssetChainFee(assetOut),
-            asset
-          })
-          const formattedThorOutFee = formatFee({
-            amount: thorOut,
-            asset: AssetRuneNative
-          })
-
-          if (!isDesktopView) {
-            return `${intl.formatMessage({
-              id: 'common.fees'
-            })}: ${formattedThorMemoFee} + ${formattedThorOutFee} + ${formattedAssetOutFee}`
-          }
-
-          return (
-            <>
-              {intl.formatMessage(
-                { id: 'deposit.withdraw.fees' },
-                { thorMemo: formattedThorMemoFee, assetOut: formattedAssetOutFee, thorOut: formattedThorOutFee }
-              )}
-            </>
-          )
-        }
+        () => 0,
+        () => 100
       )
     )
-  }, [asset, calculateAssetChainFee, fees, intl, isDesktopView])
+
+    // title
+    const txModalTitle = FP.pipe(
+      withdrawRD,
+      RD.fold(
+        () => 'deposit.withdraw.pending',
+        () => 'deposit.withdraw.pending',
+        () => 'deposit.withdraw.error',
+        () => 'deposit.withdraw.success'
+      ),
+      (id) => intl.formatMessage({ id })
+    )
+
+    const extraResult = (
+      <Styled.ExtraContainer>
+        {FP.pipe(withdrawTx, RD.toOption, (oTxHash) => (
+          <Styled.ViewTxButtonTop
+            txHash={oTxHash}
+            onClick={viewRuneTx}
+            label={intl.formatMessage({ id: 'common.tx.view' }, { assetSymbol: AssetRuneNative.symbol })}
+          />
+        ))}
+      </Styled.ExtraContainer>
+    )
+
+    return (
+      <TxModal
+        title={txModalTitle}
+        onClose={onCloseTxModal}
+        onFinish={onFinishTxModal}
+        startTime={depositStartTime}
+        txRD={withdrawRD}
+        timerValue={timerValue}
+        extraResult={extraResult}
+        extra={txModalExtraContent}
+      />
+    )
+  }, [withdrawState, onCloseTxModal, onFinishTxModal, depositStartTime, txModalExtraContent, intl, viewRuneTx])
+
+  const [showPasswordModal, setShowPasswordModal] = useState(false)
+
+  const closePasswordModal = useCallback(() => {
+    setShowPasswordModal(false)
+  }, [setShowPasswordModal])
+
+  const onClosePasswordModal = useCallback(() => {
+    // close password modal
+    closePasswordModal()
+  }, [closePasswordModal])
```

Comment:
seems this are the same methods. is there any sense why we should split it ?

Reply:
Good find, just an accident by copy / paste from other sources